subprojects/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyMetadataNotationParser.java

```diff
@@ -20,12 +20,13 @@
 import org.gradle.internal.reflect.Instantiator;
 import org.gradle.internal.typeconversion.NotationParser;
 import org.gradle.internal.typeconversion.NotationParserBuilder;
+import org.gradle.util.internal.SimpleMapInterner;
 
 public class DependencyMetadataNotationParser {
-    public static <T extends DependencyMetadata> NotationParser<Object, T> parser(Instantiator instantiator, Class<T> implementationType) {
+    public static <T extends DependencyMetadata> NotationParser<Object, T> parser(Instantiator instantiator, Class<T> implementationType, SimpleMapInterner stringInterner) {
```

Comment:
:o: Shouldn't refer to the implementation type. Same for several other places.

Reply:
That's right. I did it for a reason, that we already have a `Interner<String>` in the services, so I wanted to make sure our DI engine finds the right one. But honestly I didn't check if it would have worked with the interface type. I need to check :)

Reply:
This wouldn't be an issue if we collapsed this into a bigger service that just has a method for interning, see my comment above. I think that would make things easier for adding further  object reuse services.

Reply:
But if we collapse string interning and the module / attribute factories into one object, it would certainly reduce the verbosity of constructors, but then most consumers would only use part of it.

Reply:
The same is true for `ObjectFactory` or `ProjectLayout`. But still these methods belong closely together.